op-node/rollup/types.go

```diff
@@ -277,6 +277,20 @@ func (cfg *Config) Check() error {
 	if cfg.L2ChainID.Sign() < 1 {
 		return ErrL2ChainIDNotPositive
 	}
+	if cfg.EcotoneTime != nil {
```

Comment:
Could we instead use a function like `checkForkComesAfter(first, second *uint64)` and then run call it for the following pairs: `(regolith, canyon), (canyon, delta), (delta, ecotone)`. We're going to be adding more forks in the future & it needs to scale well.

Also we tend to return named errors & it'd be nice to add some unit tests for this check.

Reply:
Thanks for the advice, I copied the `checkFork` function from `DeployConfig`, not sure whether it's good enough or we'd like to extract it to some common place. Waiting for your advice for the next step:)

Reply:
let's pull it to a top level (but private function). The usage looks great

Reply:
Done:)